### ISSUE TITLE
TypeDoc 0.16.4: project.removeReflection does not fully remove a reflection

### ISSUE THREAD

**Issue by christopherthielen** [TypeStrong/typedoc#1176]
- [x] I have checked [issues with bug label](https://github.com/TypeStrong/typedoc/labels/bug) and found no duplicates

## Expected Behavior

Calling `ProjectReflection.removeReflection()` should remove the reflection from the project

## Actual Behavior

A reference to the removed reflection remains in the `ProjectReflection.reflections` array.  This array is still used in some typedoc code to iterate reflections.  

If `--json` is used, the reflection does not appear in the output.  However, the reflection appears in the output if `--out dist` is used.

## Steps to reproduce the bug

Create a plugin that deletes reflections, then generate HTML docs.  Note that the deleted modules appear in the module list table of contents.

`./delete-this-reflection.ts`
```ts
export class Test {
  public number = 1;
  private string = "asfff";
  method(arg: number) { }
}
```

`./delete-reflections-plugin.js`
```js
const { Converter } = require('typedoc/dist/lib/converter');
module.exports = function load(host) {
  const reflections = [];
  const app = host.application;

  app.converter.on(Converter.EVENT_CREATE_DECLARATION, (context, reflection) => {
    reflections.push(reflection);
  });

  app.converter.on(Converter.EVENT_RESOLVE_BEGIN, context => {
    reflections.forEach(reflection => context.project.removeReflection(reflection));
  });
}
```

Then run this command:
```
npm init -y
npm install typedoc@0.16.4
./node_modules/.bin/typedoc --plugin ../../../../../delete-reflections-plugin ./delete-this-reflection.ts --out dist
```

Note that the deleted Module reflection still appears in the table of contents:

![Screen Shot 2020-01-15 at 1 05 36 PM](https://user-images.githubusercontent.com/2053478/72471600-d0269400-3797-11ea-89ff-c6cbd124c484.png)

## Environment
  - Typedoc version: 0.16.4
  - Node.js version: 10.15.1
  - OS: MacOS Catalina 10.15.2

## Other details

I encountered this while trying to update `typescript-plugin-external-module-name` plugin to support 0.16.4

**Comment by Gerrit0** [comment]
Thanks for the reproduction! Should be an easy fix.

**Comment by Gerrit0** [comment]
Fixed in 0.16.6

**Comment by christopherthielen** [comment]
wow thanks! so fast :)
